src/main/java/muller/command/AddCommand.java

```diff
@@ -0,0 +1,121 @@
+package muller.command;
+
+import java.time.LocalDate;
+
+import muller.storage.Storage;
+import muller.task.Task;
+import muller.task.TaskList;
+import muller.ui.Ui;
+
+/**
+ * Command to add a task (Todo, Deadline, or Event) to the task list.
+ */
+public class AddCommand extends Command {
+    private String[] inputs;
+    private String type;
+
+    /**
+     * Constructs an AddCommand with the specified inputs and task type.
+     *
+     * @param inputs The command inputs.
+     * @param type   The type of task to add (T, D, E).
+     */
+    public AddCommand(String[] inputs, String type) {
+        this.inputs = inputs;
+        this.type = type;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws MullerException {
+        if (inputs.length < 2) {
+            throw new MullerException(type.equals("T") ? "Todo what?"
+                    : type.equals("D") ? "Deadline for what?" : "Event for what?");
+        }
+        String details = inputs[1].trim();
+        Task task;
+
+        switch (type) {
+        case "T":
+            task = new Task(details);
+            task.setType("T");
+            break;
+        case "D":
+            task = parseDeadline(details);
+            break;
+        case "E":
+            task = parseEvent(details);
+            break;
+        default:
+            throw new MullerException("Unknown task type!");
+        }
+
+        tasks.addTask(task);
+        ui.showLine();
+        System.out.println("Got it. I've added this task:");
```

Comment:
Would it be better to specify a method for printing statements in the UI class instead?